### PR TITLE
Add upsert logic for sql server

### PR DESCRIPTION
**Pull request by JeppeBlixen**

### CODE REVIEW

**Comment by JeppeBlixen** [comment]
First step to create common sql and delta merge query

**Comment by LauJohansson** [comment]
@JeppeBlixen - we use black, isort and flake8. You can wither add them to pre-commits or use File Watchers in your IDE.

File: src/atc/etl/loaders/UpsertLoader.py [hunk 1]
```diff
@@ -51,6 +51,9 @@ def save(self, df_source: DataFrame):
             df = df.filter(
                 " AND ".join(f"({col} is NOT NULL)" for col in self.params.join_cols)
             )
+            df = df.filter(
+                " AND ".join(f"({col} is NOT NULL)" for col in self.params.join_cols)
+            )
```
Comment (LauJohansson) [inline src/atc/etl/loaders/UpsertLoader.py:56]: This seems redundant...

File: src/atc/sql/SqlServer.py [hunk 2]
```diff
@@ -100,6 +104,48 @@ def write_table_by_name(
...
+            source_table_name=staging_table_name,
+            join_cols=join_cols,
+            insert_cols=df_source.columns,
+            update_cols=df_source.columns,
+        )
+
+        # Merge query is combined with drop staging table
+        # to ensure the drop of the staging table in case
+        # the problem stops in the midle of the upsert
+        mergeQuery += f" DROP TABLE {staging_table_name};"
```
Comment (LauJohansson) [inline src/atc/sql/SqlServer.py:145]: should the mergestatement + drop table be in a transation statement? https://docs.microsoft.com/en-us/sql/t-sql/language-elements/begin-transaction-transact-sql?view=sql-server-ver16

**Review by LauJohansson — CHANGES_REQUESTED** [review]

File: src/atc/sql/SqlServer.py (hunk 2 above)
Comment (JeppeBlixen) [reply src/atc/sql/SqlServer.py:145]: I want to be sure that the table is dropped. I have seen examples where the job is stoppd in the middle of the merge(ex when stopping a stream). In this case the case the staging tables would pile up, as this happens.

**Review by LauJohansson — APPROVED** [review]

**Pull request merged**